File: app/src/main/java/org/wikipedia/util/ActiveTimer.java
package org.wikipedia.util;

import java.util.concurrent.TimeUnit;

public class ActiveTimer {
    private long startMillis;
    private long pauseMillis;

    public ActiveTimer() {
        reset();
    }

    public void reset() {
        startMillis = System.currentTimeMillis();
        pauseMillis = startMillis;
    }

    public void pause() {
        pauseMillis = System.currentTimeMillis();
    }

    public void resume() {
        startMillis -= (System.currentTimeMillis() - pauseMillis);
    }

    public int getElapsedSec() {
        return (int) TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis() - startMillis);
    }
}

File: app/src/main/java/org/wikipedia/util/ActiveTimer.kt
package org.wikipedia.util

import java.util.concurrent.TimeUnit

class ActiveTimer {
    private var startMillis: Long = 0
    private var pauseMillis: Long = 0
    private var isPaused = false

    init {
        reset()
    }

    fun reset() {
        startMillis = System.currentTimeMillis()
        pauseMillis = startMillis
    }

    fun pause() {
        pauseMillis = System.currentTimeMillis()
        isPaused = true
    }

    fun resume() {
        startMillis += System.currentTimeMillis() - pauseMillis
        isPaused = false
    }

    val elapsedMillis: Long
        get() = if (isPaused) pauseMillis - startMillis else System.currentTimeMillis() - startMillis

    val elapsedSec: Int
        get() = TimeUnit.MILLISECONDS.toSeconds(elapsedMillis).toInt()
}
